src/functions/__tests__/fast-login.test.ts

```diff
@@ -220,10 +248,55 @@ describe("Fast Login handler", () => {
       const result = await makeFastLoginHandler({
         ...httpHandlerInputMocks,
         input: req,
-        fnLollipopClient: mockedFnLollipopClient
+        fnLollipopClient: mockedFnLollipopClient,
+        blobService: mockBlobService
       })();
       expect(getAssertionMock).toBeCalled();
+      expect(mockUpsertBlobFromObject).not.toBeCalled();
       expect(result).toEqual(E.left(new H.HttpError(expectedErrorMessage)));
     }
   );
+
+  it.each`
+    WHEN                                      | upsertBlobFromObjectResult                    | expectedErrorMessage                                                 | resolve
+    ${"the blob storage returns no result"}   | ${E.right(O.none)}                            | ${"The audit log was not saved"}                                     | ${true}
+    ${"the blob wasn't created"}              | ${E.right(O.some({ created: false }))}        | ${"The audit log was not saved"}                                     | ${true}
+    ${"the blob sorage returns an error"}     | ${E.left(new Error("Storage Error Message"))} | ${"An error occurred saving the audit log: [Storage Error Message]"} | ${true}
+    ${"the blob sorage unexpectedly rejects"} | ${new Error("Unexpected error")}              | ${"Unexpected error: [Unexpected error]"}                            | ${false}
+  `(
+    `GIVEN a valid LolliPoP request
+     WHEN $WHEN
+     THEN an Internal Server Error response is returned`,
+    async ({ upsertBlobFromObjectResult, expectedErrorMessage, resolve }) => {
+      resolve
+        ? mockUpsertBlobFromObject.mockResolvedValueOnce(
+            upsertBlobFromObjectResult
+          )
+        : mockUpsertBlobFromObject.mockRejectedValueOnce(
+            upsertBlobFromObjectResult
+          );
+      const req: H.HttpRequest = {
+        ...H.request("https://api.test.it/"),
+        headers: {
+          ...validLollipopHeaders,
+          ...validFastLoginAdditionalHeaders
+        }
+      };
+      const result = await makeFastLoginHandler({
+        ...httpHandlerInputMocks,
+        input: req,
+        fnLollipopClient: mockedFnLollipopClient,
+        blobService: mockBlobService
+      })();
+      expect(getAssertionMock).toBeCalled();
+      expect(mockUpsertBlobFromObject).toBeCalled();
```

Comment:
I think we can enrich this test including the arguments passed to the `upsertBlobFromObject` function
```suggestion
      expect(mockUpsertBlobFromObject).toBeCalled();
      expect(mockUpsertBlobFromObject).toBeCalledWith(
        {},
        "logs",
        expect.stringContaining(
          validLollipopHeaders["x-pagopa-lollipop-user-id"]
        ),
        expect.objectContaining({
          assertion_xml: aSAMLResponse,
          client_ip: validFastLoginAdditionalHeaders["x-pagopa-lv-client-ip"]
        })
      );
```

Reply:
Fixed in [fe0d4b2](https://github.com/pagopa/io-functions-fast-login/pull/15/commits/fe0d4b260714df1876e67003dfc749d7410a4373)